src/plugins/intel_gpu/tests/test_cases/extract_image_patches_gpu_test.cpp

```diff
@@ -507,29 +508,47 @@ TEST(extract_image_patches_gpu, basic5) {
 
     auto input = engine.allocate_memory({ data_types::f32, format::bfyx, { batch, depth, in_cols, in_rows } });
 
-    std::vector<float> inputVals(batch * depth * in_rows * in_cols);
-    std::generate(inputVals.begin(), inputVals.end(), []() {
-        static float n = 1;
-        return n++;
-    });
+    std::vector<T> inputVals(batch * depth * in_rows * in_cols);
+    float n = 1;
+    for (auto& val : inputVals) {
+        val = n++;
+    }
 
     set_values(input, inputVals);
 
     topology topology;
     topology.add(input_layout("Input0", input->get_layout()));
     topology.add(extract_image_patches("extract_image_patches", "Input0", sizes, strides, rates, auto_pad, output_shape));
 
-    network network(engine, topology);
-    network.set_input_data("Input0", input);
-    auto outputs = network.execute();
+    cldnn::network::ptr network;
+
+    if (is_caching_test()) {
+        membuf mem_buf;
+        {
+            cldnn::network _network(engine, topology);
+            std::ostream out_mem(&mem_buf);
+            BinaryOutputBuffer ob = BinaryOutputBuffer(out_mem);
+            _network.save(ob);
+        }
+        {
+            std::istream in_mem(&mem_buf);
+            BinaryInputBuffer ib = BinaryInputBuffer(in_mem, engine);
+            network = std::make_shared<cldnn::network>(ib, get_test_stream_ptr(), engine);
```

Comment:
Do we really need stream::ptr arg in this network contructor? Can't it call create_stream() internally?

Reply:
I added [GPU] tags. Thank you.